// @esri/calcite-components/dist/cjs/bs-55afe249.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.7.0
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-8637bb2e.js');
require('./index-5cedda26.js');
require('./form-27befe54.js');
require('./dom-be7b3b5e.js');
require('./guid-a8269edb.js');
require('./resources-4717d162.js');
require('./interactive-3bb6deb3.js');
require('./browser-c9faecb4.js');
require('./key-a0314f25.js');
require('./label-dc984ee1.js');
require('./loadable-9f17151a.js');
require('./locale-29f97b64.js');
require('./observers-1af2e17e.js');
require('./focusTrapComponent-917ed18b.js');
require('./math-a50a7a62.js');
require('./t9n-a92a13a7.js');

// Bosnian [bs]
var locale = {
  name: 'bs',
  weekdays: 'nedjelja_ponedjeljak_utorak_srijeda_četvrtak_petak_subota'.split('_'),
  months: 'januar_februar_mart_april_maj_juni_juli_august_septembar_oktobar_novembar_decembar'.split('_'),
  weekStart: 1,
  weekdaysShort: 'ned._pon._uto._sri._čet._pet._sub.'.split('_'),
  monthsShort: 'jan._feb._mar._apr._maj._jun._jul._aug._sep._okt._nov._dec.'.split('_'),
  weekdaysMin: 'ne_po_ut_sr_če_pe_su'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D. MMMM YYYY',
    LLL: 'D. MMMM YYYY H:mm',
    LLLL: 'dddd, D. MMMM YYYY H:mm'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports.default = locale;
