// @esri/calcite-components/dist/cjs/calcite-input-time-zone.cjs.entry.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.7.0
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-5cedda26.js');
const label = require('./label-dc984ee1.js');
const interactive = require('./interactive-3bb6deb3.js');
const locale = require('./locale-29f97b64.js');
const t9n = require('./t9n-a92a13a7.js');
const loadable = require('./loadable-9f17151a.js');
const form = require('./form-27befe54.js');
require('./dom-be7b3b5e.js');
require('./guid-a8269edb.js');
require('./resources-4717d162.js');
require('./browser-c9faecb4.js');
require('./key-a0314f25.js');
require('./observers-1af2e17e.js');

const hourToMinutes = 60;
function getUserTimeZoneOffset() {
  const localDate = new Date();
  return localDate.getTimezoneOffset() * -1;
}
function getFallbackTimeZoneGroups() {
  const timeZoneOffsets = [
    -11, -10, -9.5, -9, -8, -7, -6, -5, -4, -3, -2.5, -2, -1, 0, 1, 2, 3, 3.5, 4, 4.5, 5, 6, 6.5, 7, 8, 8.75, 9, 9.5,
    10, 10.5, 11, 12, 12.75, 13, 14,
  ];
  return timeZoneOffsets.map((offset) => {
    return {
      offsetValue: offset * hourToMinutes,
      offsetLabel: toGMTLabel(offset),
    };
  });
}
/**
 * Exported for testing-purposes only
 *
 * @internal
 */
function toGMTLabel(offsetInHours) {
  return `GMT${offsetInHours === 0 ? "" : offsetInHours.toLocaleString("en", { signDisplay: "always" })}`;
}
let timeZoneGeneration;
async function generateTimeZoneGroups() {
  if (timeZoneGeneration) {
    return timeZoneGeneration;
  }
  timeZoneGeneration = Promise.resolve(getFallbackTimeZoneGroups());
  return timeZoneGeneration;
}

const inputTimeZoneCss = ":host([disabled]){cursor:default;-webkit-user-select:none;user-select:none;opacity:var(--calcite-ui-opacity-disabled)}:host([disabled]) *,:host([disabled]) ::slotted(*){pointer-events:none}:host{display:block}:host([hidden]){display:none}[hidden]{display:none}:host([disabled]) ::slotted([calcite-hydrated][disabled]),:host([disabled]) [calcite-hydrated][disabled]{opacity:1}::slotted(input[slot=hidden-form-input]){margin:0 !important;opacity:0 !important;outline:none !important;padding:0 !important;position:absolute !important;inset:0 !important;transform:none !important;-webkit-appearance:none !important;z-index:-1 !important}";

const InputTimeZone = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.calciteInputTimeZoneBeforeClose = index.createEvent(this, "calciteInputTimeZoneBeforeClose", 6);
    this.calciteInputTimeZoneBeforeOpen = index.createEvent(this, "calciteInputTimeZoneBeforeOpen", 6);
    this.calciteInputTimeZoneChange = index.createEvent(this, "calciteInputTimeZoneChange", 6);
    this.calciteInputTimeZoneClose = index.createEvent(this, "calciteInputTimeZoneClose", 6);
    this.calciteInputTimeZoneOpen = index.createEvent(this, "calciteInputTimeZoneOpen", 6);
    this.setComboboxRef = (el) => {
      this.comboboxEl = el;
    };
    this.onComboboxBeforeClose = (event) => {
      event.stopPropagation();
      this.calciteInputTimeZoneBeforeClose.emit();
    };
    this.onComboboxBeforeOpen = (event) => {
      event.stopPropagation();
      this.calciteInputTimeZoneBeforeOpen.emit();
    };
    this.onComboboxChange = (event) => {
      event.stopPropagation();
      const combobox = event.target;
      const selected = this.timeZoneGroups.find(({ offsetValue }) => combobox.value === `${offsetValue}`);
      const selectedValue = `${selected.offsetValue}`;
      if (this.value === selectedValue) {
        return;
      }
      this.value = selectedValue;
      this.selectedTimeZoneGroup = selected;
      this.calciteInputTimeZoneChange.emit();
    };
    this.onComboboxClose = (event) => {
      event.stopPropagation();
      this.open = false;
      this.calciteInputTimeZoneClose.emit();
    };
    this.onComboboxOpen = (event) => {
      this.open = true;
      event.stopPropagation();
      this.calciteInputTimeZoneOpen.emit();
    };
    this.disabled = false;
    this.form = undefined;
    this.messages = undefined;
    this.messageOverrides = undefined;
    this.name = undefined;
    this.open = false;
    this.overlayPositioning = "absolute";
    this.required = false;
    this.scale = "m";
    this.value = undefined;
    this.defaultMessages = undefined;
    this.effectiveLocale = "";
  }
  onMessagesChange() {
    /* wired up by t9n util */
  }
  valueWatcher(value) {
    if (this.comboboxEl && this.comboboxEl.value !== value) {
      this.comboboxEl.value = value;
    }
  }
  //--------------------------------------------------------------------------
  //
  //  Public Methods
  //
  //--------------------------------------------------------------------------
  async setFocus() {
    await loadable.componentFocusable(this);
    await this.comboboxEl.setFocus();
  }
  effectiveLocaleWatcher() {
    t9n.updateMessages(this, this.effectiveLocale);
  }
  //--------------------------------------------------------------------------
  //
  //  Private Methods
  //
  //--------------------------------------------------------------------------
  onLabelClick() {
    this.setFocus();
  }
  // --------------------------------------------------------------------------
  //
  //  Lifecycle
  //
  //--------------------------------------------------------------------------
  connectedCallback() {
    form.connectForm(this);
    label.connectLabel(this);
    locale.connectLocalized(this);
    t9n.connectMessages(this);
  }
  disconnectedCallback() {
    form.disconnectForm(this);
    label.disconnectLabel(this);
    locale.disconnectLocalized(this);
    t9n.disconnectMessages(this);
  }
  async componentWillLoad() {
    loadable.setUpLoadableComponent(this);
    await t9n.setUpMessages(this);
    const timeZoneGroups = await generateTimeZoneGroups();
    this.timeZoneGroups = timeZoneGroups;
    const offsetToMatch = this.value ?? getUserTimeZoneOffset();
    this.selectedTimeZoneGroup = timeZoneGroups.find(({ offsetValue }) => 
    // intentional == to match string to number
    offsetValue == offsetToMatch);
    const selectedValue = `${this.selectedTimeZoneGroup.offsetValue}`;
    form.afterConnectDefaultValueSet(this, selectedValue);
    this.value = selectedValue;
  }
  componentDidLoad() {
    loadable.setComponentLoaded(this);
  }
  componentDidRender() {
    interactive.updateHostInteraction(this);
  }
  render() {
    return (index.h(index.Host, null, index.h("calcite-combobox", { clearDisabled: true, disabled: this.disabled, label: this.messages.chooseTimeZone, lang: this.effectiveLocale, onCalciteComboboxBeforeClose: this.onComboboxBeforeClose, onCalciteComboboxBeforeOpen: this.onComboboxBeforeOpen, onCalciteComboboxChange: this.onComboboxChange, onCalciteComboboxClose: this.onComboboxClose, onCalciteComboboxOpen: this.onComboboxOpen, open: this.open, overlayPositioning: this.overlayPositioning, scale: this.scale, selectionMode: "single",
      // eslint-disable-next-line react/jsx-sort-props -- ref should be last so node attrs/props are in sync (see https://github.com/Esri/calcite-design-system/pull/6530)
      ref: this.setComboboxRef }, this.timeZoneGroups.map((group) => {
      const selected = this.selectedTimeZoneGroup === group;
      const label = group.offsetLabel;
      const value = group.offsetValue;
      return (index.h("calcite-combobox-item", { key: label, selected: selected, textLabel: label, value: `${value}` }));
    })), index.h(form.HiddenFormInputSlot, { component: this })));
  }
  static get delegatesFocus() { return true; }
  static get assetsDirs() { return ["assets"]; }
  get el() { return index.getElement(this); }
  static get watchers() { return {
    "messageOverrides": ["onMessagesChange"],
    "value": ["valueWatcher"],
    "effectiveLocale": ["effectiveLocaleWatcher"]
  }; }
};
InputTimeZone.style = inputTimeZoneCss;

exports.calcite_input_time_zone = InputTimeZone;
