// @esri/calcite-components/dist/esm/bs-41a2ac4e.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.7.0
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-7f52a0b1.js';
import './index-165cd388.js';
import './form-c73d3da5.js';
import './dom-79f84f98.js';
import './guid-3f32a87e.js';
import './resources-4b333cce.js';
import './interactive-5801a6e6.js';
import './browser-612db93f.js';
import './key-120fe23c.js';
import './label-b88198aa.js';
import './loadable-d814078e.js';
import './locale-b09d220a.js';
import './observers-200147a2.js';
import './focusTrapComponent-0e7ce2bc.js';
import './math-4e0ea91f.js';
import './t9n-9a6203a1.js';

// Bosnian [bs]
var locale = {
  name: 'bs',
  weekdays: 'nedjelja_ponedjeljak_utorak_srijeda_četvrtak_petak_subota'.split('_'),
  months: 'januar_februar_mart_april_maj_juni_juli_august_septembar_oktobar_novembar_decembar'.split('_'),
  weekStart: 1,
  weekdaysShort: 'ned._pon._uto._sri._čet._pet._sub.'.split('_'),
  monthsShort: 'jan._feb._mar._apr._maj._jun._jul._aug._sep._okt._nov._dec.'.split('_'),
  weekdaysMin: 'ne_po_ut_sr_če_pe_su'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D. MMMM YYYY',
    LLL: 'D. MMMM YYYY H:mm',
    LLLL: 'dddd, D. MMMM YYYY H:mm'
  }
};
dayjs.locale(locale, null, true);

export default locale;
