// @esri/calcite-components/dist/esm/nb-1b8c3a74.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.7.0
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-7f52a0b1.js';
import './index-165cd388.js';
import './form-c73d3da5.js';
import './dom-79f84f98.js';
import './guid-3f32a87e.js';
import './resources-4b333cce.js';
import './interactive-5801a6e6.js';
import './browser-612db93f.js';
import './key-120fe23c.js';
import './label-b88198aa.js';
import './loadable-d814078e.js';
import './locale-b09d220a.js';
import './observers-200147a2.js';
import './focusTrapComponent-0e7ce2bc.js';
import './math-4e0ea91f.js';
import './t9n-9a6203a1.js';

// Norwegian Bokmål [nb]
var locale = {
  name: 'nb',
  weekdays: 'søndag_mandag_tirsdag_onsdag_torsdag_fredag_lørdag'.split('_'),
  weekdaysShort: 'sø._ma._ti._on._to._fr._lø.'.split('_'),
  weekdaysMin: 'sø_ma_ti_on_to_fr_lø'.split('_'),
  months: 'januar_februar_mars_april_mai_juni_juli_august_september_oktober_november_desember'.split('_'),
  monthsShort: 'jan._feb._mars_april_mai_juni_juli_aug._sep._okt._nov._des.'.split('_'),
  ordinal: function ordinal(n) {
    return n + ".";
  },
  weekStart: 1,
  yearStart: 4,
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D. MMMM YYYY',
    LLL: 'D. MMMM YYYY [kl.] HH:mm',
    LLLL: 'dddd D. MMMM YYYY [kl.] HH:mm'
  },
  relativeTime: {
    future: 'om %s',
    past: '%s siden',
    s: 'noen sekunder',
    m: 'ett minutt',
    mm: '%d minutter',
    h: 'en time',
    hh: '%d timer',
    d: 'en dag',
    dd: '%d dager',
    M: 'en måned',
    MM: '%d måneder',
    y: 'ett år',
    yy: '%d år'
  }
};
dayjs.locale(locale, null, true);

export default locale;
